client/src/controllers/LinkController.ts

```diff
@@ -19,7 +19,7 @@ export class LinkController extends Controller<HTMLElement> {
   }
```

Comment:
I would be really great if we can also add some JSDoc to this controller, here's a rough idea, based on my understanding of what this does.


```ts
/**
 * Adds the ability for the controlled element to reflect the URL's query parameters
 * from an event or the current URL into the window.location so that dynamic updates
 * to listings can be available on refreshing or other internal links.
 * 
 * @example - Reflecting the URL's query parameters
 * <a href="/?q=1" data-controller="w-link" data-w-link-reflect-keys-value="q"></a>
 */
```

Additionally, adding JSDoc to the `declare` statements helps when editing the code, here's a rough guess.


```ts
  /** Attribute on the controlled element containing the URL string. */
  declare attrNameValue: string;
  /** URL param keys that will be kept in the current location's URL. */
  declare preserveKeysValue: string[];
  /** URL param keys to be added to the location URL from the source URL. */
  declare reflectKeysValue: string[];
```